pkg/reconciler/revision/resources/deploy_test.go

```diff
@@ -680,28 +902,99 @@ func TestMakePodSpec(t *testing.T) {
 		),
 		want: podSpec(
 			[]corev1.Container{
-				userContainer(
+				servingContainer(
+					func(container *corev1.Container) {
+						container.Image = "busybox@sha256:deadbeef"
+					},
+					withContainer(),
+					withEnvVar("K_CONFIGURATION", "cfg"),
+					withEnvVar("K_SERVICE", "svc"),
+				),
+				queueContainer(
+					withEnvVar("CONTAINER_CONCURRENCY", "1"),
+					withEnvVar("SERVING_SERVICE", "svc"),
+				),
+			}),
+	}, {
+		name: "complex pod spec for multiple containers with container data to all containers",
+		rev: revision("bar", "foo",
+			withContainers([]corev1.Container{
+				{
+					Name:  servingContainerName,
+					Image: "busybox",
+					Ports: []corev1.ContainerPort{{
+						ContainerPort: 8080,
+					}},
+				}, {
+					Name:  sidecarContainerName,
+					Image: "ubuntu",
+				}, {
+					Name:  "sidecar-container-2",
+					Image: "alpine",
+				},
+			}),
```

Comment:
nit:

```suggestion
			withContainers([]corev1.Container{{
					Name:  servingContainerName,
					Image: "busybox",
					Ports: []corev1.ContainerPort{{
						ContainerPort: 8080,
					}},
				}, {
					Name:  sidecarContainerName,
					Image: "ubuntu",
				}, {
					Name:  "sidecar-container-2",
					Image: "alpine",
				}}),
```
Should collapse the indentation a bit.